packages/services/api/src/modules/schema/resolvers/FailedSchemaCheck.ts

```diff
@@ -0,0 +1,57 @@
+import { ContractsManager } from '../providers/contracts-manager';
+import { SchemaCheckManager } from '../providers/schema-check-manager';
+import { SchemaManager } from '../providers/schema-manager';
+import type { FailedSchemaCheckResolvers } from './../../../__generated__/types.next';
+
+export const FailedSchemaCheck: FailedSchemaCheckResolvers = {
+  schemaVersion: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getSchemaVersion(schemaCheck);
+  },
+  safeSchemaChanges: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getSafeSchemaChanges(schemaCheck);
+  },
+  breakingSchemaChanges: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getBreakingSchemaChanges(schemaCheck);
+  },
+  compositionErrors: schemaCheck => {
+    return schemaCheck.schemaCompositionErrors;
+  },
+  hasSchemaCompositionErrors: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getHasSchemaCompositionErrors(schemaCheck);
+  },
+  hasSchemaChanges: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getHasSchemaChanges(schemaCheck);
+  },
+  hasUnapprovedBreakingChanges: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getHasUnapprovedBreakingChanges(schemaCheck);
+  },
+  webUrl: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaManager).getSchemaCheckWebUrl({
+      schemaCheckId: schemaCheck.id,
+      targetId: schemaCheck.targetId,
+    });
+  },
+  canBeApproved: async (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaManager).getFailedSchemaCheckCanBeApproved(schemaCheck);
+  },
+  canBeApprovedByViewer: async (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaManager).getFailedSchemaCheckCanBeApprovedByViewer(schemaCheck);
+  },
+  contractChecks: (schemaCheck, _, { injector }) => {
+    return injector.get(ContractsManager).getContractsChecksForSchemaCheck(schemaCheck);
+  },
+  previousSchemaSDL: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getPreviousSchemaSDL(schemaCheck);
+  },
+  conditionalBreakingChangeMetadata: (schemaCheck, _, { injector }) => {
+    return injector.get(SchemaCheckManager).getConditionalBreakingChangeMetadata(schemaCheck);
+  },
+  schemaPolicyErrors: ({ schemaPolicyErrors }, _arg, _ctx) => {
+    // FIXME: @eddeee888 check why this doesn't have runtime error
+    return schemaPolicyErrors as any;
+  },
+  schemaPolicyWarnings: ({ schemaPolicyWarnings }, _arg, _ctx) => {
+    // FIXME: @eddeee888 check why this doesn't have runtime error
+    return schemaPolicyWarnings as any;
+  },
```

Comment:
The resolvers that are returning `SchemaPolicyWarningConnection` seem to detect potential runtime error because there's a difference between the [mapper](https://github.com/kamilkisiela/graphql-hive/blob/2af71a2bc0d74b0a46cf082a4b67566069dc27b9/packages/services/api/src/modules/schema/providers/models/shared.ts#L81-L89) vs the type from [storage model](https://github.com/kamilkisiela/graphql-hive/blob/7c476f941dd935460aba7e6e6b0b2079032f89d3/packages/services/storage/src/schema-change-model.ts#L797-L804):

| Field | Mapper | Storage | Compatible |
| -- | -- | -- | -- |
| message | `string` | `string` | ✅ |
| source | `string` | N/A | ❌ |
| line | `number` | `numer` or `null` or `undefined` | ❌ |
| column | `number` | `numer` or `null` or `undefined` | ❌ |
| ruleId | `string` | `string` or `null` | ❌ |
| endLine | `number` or `null` | `number` or `null` or `undefined` | ❌ |
| endColumn | `number` or `null` | `number` or `null` or `undefined` | ❌ |

If I try to change the mapper to the storage type, it works but [SchemaPolicyWarningConnection](https://github.com/kamilkisiela/graphql-hive/pull/5283/files#diff-8776f3d9c5421f59077d3af32a5fa8169786fc231630ea3f4ce5c7b1573ff13fR9-R12) would have type errors.
And changing the Storage zod model to match the mapper's might cause issue at runtime.

Either way, I think we'd need to fix some runtime implementation, and I'd need to look deeper to see which way approach is better. So, very open for suggestions how to fix this issue 🙂

Another option is to keep the FIXME and keep going as I don't like holding a big PR like this open for too long as it might have conflicts

Reply:
This is fixed in https://github.com/kamilkisiela/graphql-hive/pull/5283/commits/d4a46a246bc08a2216dd6a3edddd38d209de5ec1 by updating the `SchemaCheckWarning` type (used as mapper)